src/Sulu/Bundle/WebsiteBundle/Tests/Unit/Sulu/Bundle/WebsiteBundle/Cache/CacheClearerTest.php

```diff
@@ -0,0 +1,143 @@
+     */
+    private $requestStack;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var CacheManager|null
+     */
+    private $cacheManager;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function setUp(): void
+    {
+        $this->eventDispatcher = $this->prophesize(EventDispatcherInterface::class);
+        $this->cacheManager = $this->prophesize(CacheManager::class);
+        $this->requestStack = $this->prophesize(RequestStack::class);
+        $this->request = $this->prophesize(Request::class);
+        $this->filesystem = $this->prophesize(Filesystem::class);
+    }
+
+    public function testClearOnFilesystem(): void
+    {
+        $expectedPath = __DIR__ . '/var/cache/common/test/http_cache';
+
+        $this->filesystem->exists($expectedPath)->willReturn(true)->shouldBeCalled();
+        $this->filesystem->remove($expectedPath)->shouldBeCalled();
+
+        $this->eventDispatcher->dispatch(
+            Argument::type(CacheClearEvent::class),
+            Events::CACHE_CLEAR
+        )->shouldBeCalled();
+
+        $cacheClearer = $this->createCacheClearer();
+        $cacheClearer->clear();
+    }
+
+    public function testClearOnFilesystemNotExist(): void
+    {
+        $expectedPath = __DIR__ . '/var/cache/common/test/http_cache';
+
+        $this->filesystem->exists($expectedPath)->willReturn(false)->shouldBeCalled();
+        $this->filesystem->remove($expectedPath)->shouldNotBeCalled();
+
+        $this->eventDispatcher->dispatch(
+            Argument::type(CacheClearEvent::class),
+            Events::CACHE_CLEAR
+        )->shouldBeCalled();
+
+        $cacheClearer = $this->createCacheClearer();
+        $cacheClearer->clear();
+    }
+
+    public function testClearOnCacheManagerNoRequest(): void
+    {
+        $this->requestStack->getCurrentRequest()->shouldBeCalled();
+
+        $this->cacheManager->supportsInvalidate()->willReturn(true);
+        $this->cacheManager->invalidateDomain(Argument::any())->shouldNotBeCalled();
+
+        $cacheClearer = $this->createCacheClearer(true);
+        $cacheClearer->clear();
+    }
+
+    public function testClearOnCacheManagerWithRequest(): void
+    {
+        $this->requestStack->getCurrentRequest()
+            ->willReturn($this->request->reveal())
+            ->shouldBeCalled();
+
+        $this->request->getHost()
+            ->willReturn('sulu.io')
+            ->shouldBeCalled();
+
+        $this->cacheManager->supportsInvalidate()->willReturn(true);
+        $this->cacheManager->invalidateDomain('sulu.io')->shouldBeCalled();
+
+        $this->eventDispatcher->dispatch(
+            Argument::type(CacheClearEvent::class),
+            Events::CACHE_CLEAR
+        )->shouldBeCalled();
+
+        $cacheClearer = $this->createCacheClearer(true);
+        $cacheClearer->clear();
+    }
+
+    private function createCacheClearer(bool $withCacheManager = false): CacheClearer
```

Comment:
Why don't we pass the `cacheManager` itself instead of a boolean? I think that `$this->createCacheClearer($cacheManager)` would be clearer than `$this->createCacheClearer(true)`. In that case I would also create the `CacheManager` prophecy only in the tests that require it. Then we have only the stuff that is shared among every test in the `setUp`.

Reply:
Sounds good. Did change the test.